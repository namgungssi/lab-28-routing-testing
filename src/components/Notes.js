'use strict';



import React from 'react';
import NoteForm from './NoteCreateForm';
import {saveNote} from '../lib/helpers';



class Notes extends React.Component {
  constructor(props) {
    super(props);
    this.handleNewNote = this.handleNewNote.bind(this);
  }

  componentDidMount() {
    console.log("__E_STATE__", this.state);
  }

  handleNewNote(note) {
    saveNote(note);
    this.props.handler(note):
    location.reload();
  }

  render() {
    return (
      <div id="noteWrapper">
        <h3>Create Note</h3>
        <NoteForm handler={this.handleNewNote}/>
        </div>
    );
  }
}


export default Notes;
