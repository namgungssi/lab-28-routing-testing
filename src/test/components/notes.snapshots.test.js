'use strict';



import React from 'react';
import renderer from 'react-test-renderer';
import Notes from '../../components/Notes';



describe('<Notes>', () => {
  test('renders with a clean state', () => {

    let app = {
        state: {
            notes:[]
        },
        setState: (data) => app.state = Object.assign({}, app.state, data)
    }

    const component = renderer.create(
      <Notes app={app} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('renders notes with the correct titles', () => {
    let app = {
        state: {
            notes: [
              { title:'Note1', content:'Note1 content'},
              { title:'Note2', content:'Note2 content'}
            ]
        },
        setState: (data) => app.state = Object.assign({}, app.state, data)
    }


    const component = renderer.create(
      <Notes app={app} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
