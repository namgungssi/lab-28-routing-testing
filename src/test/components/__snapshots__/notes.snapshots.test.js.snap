// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notes> renders notes with the correct titles 1`] = `
<div
  id="noteWrapper"
>
  <h3>
    Create a Note
  </h3>
  <form
    id="noteForm"
    onSubmit={[Function]}
  >
    <input
      className="createNoteInput"
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Note Title"
      type="text"
    />
    <textarea
      className="createNoteInput"
      cols="20"
      id="content"
      name="content"
      onChange={[Function]}
      placeholder="Content"
      rows="5"
      type="text"
    />
    <button
      className="createNoteInput"
      id="submitButton"
      type="submit"
    >
      Add Note
    </button>
  </form>
</div>
`;

exports[`<Notes> renders with a clean state 1`] = `
<div
  id="noteWrapper"
>
  <h3>
    Create a Note
  </h3>
  <form
    id="noteForm"
    onSubmit={[Function]}
  >
    <input
      className="createNoteInput"
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Note Title"
      type="text"
    />
    <textarea
      className="createNoteInput"
      cols="20"
      id="content"
      name="content"
      onChange={[Function]}
      placeholder="Content"
      rows="5"
      type="text"
    />
    <button
      className="createNoteInput"
      id="submitButton"
      type="submit"
    >
      Add Note
    </button>
  </form>
</div>
`;
